### DISCUSSION TITLE
Any example how to implement it with socket.io?

### DISCUSSION THREAD

**Discussion by volarname** [ueberdosis/hocuspocus#233] [Q&A]
I have tiptap v1 collaboration server written using `github.com/socketio/socket.io`.
A lot of custom events, for example for locking article:
```
socket.on('articleJoin', ...);
socket.on('articleLeave', ...);
socket.on('cursorChange', ...);
socket.on('update', ...);
``` 
 any example how to implement hocuspocus + tiptap v2 + cursor with small change in socket.io code, so collaboration will transfer and custom events will work on same websocket?

**Comment by hanspagel** [comment]
Nah, we won’t support socket.io. Socket.io was great when browsers didn’t have proper support for WebSockets, but those times are over. Now Socket.io has more limitations than it’s helpful (in our use case). See also: #95
